onnx/defs/quantization/defs.cc

```diff
@@ -81,17 +83,27 @@ ONNX_OPERATOR_SET_SCHEMA(
             static_cast<int64_t>(0))
         .Attr(
             "output_dtype",
-            "(Optional) The output data type. If not supplied, the output data type is inferred from `y_zero_point` data type (`T2`). "
+            "(Optional) The output data type. If not supplied, the output data type is inferred from `y_zero_point` data type (`T3`). "
             "If neither `output_dtype` nor `y_zero_point` are supplied, output data type is uint8. "
-            "If both `output_dtype` and `y_zero_point` are specified, `output_dtype` must be `T2`.",
+            "If both `output_dtype` and `y_zero_point` are specified, `output_dtype` must be `T3`.",
             AttributeProto::INT,
             static_cast<int64_t>(0))
+        .Attr(
+            "precision",
+            "(Optional) The precision of the division operation between `x` and `y_scale`. If not provided, "
+            "it will be the same as the type of `y_scale`.",
+            AttributeProto::INT,
+            OPTIONAL_VALUE)
         .TypeConstraint(
             "T1",
             {"tensor(float)", "tensor(float16)", "tensor(bfloat16)", "tensor(int32)"},
             "The type of the input 'x'.")
         .TypeConstraint(
             "T2",
+            {"tensor(float)", "tensor(float16)", "tensor(bfloat16)", "tensor(int32)"},
```

Comment:
Any reason T1 and T2 are defined as different TypeConstraints? From my understanding, the TypeConstraint only defines the range of types a certain input can be, and T1 and T2 seem to point to the same range of types.

Reply:
This is saying they can be bound to different types in a single instance

Reply:
@shubhambhokare1 `CastLike` is an example where something similar is used.